src/TrussTopOptProblems/buckling.jl

```diff
@@ -86,19 +149,36 @@ function buckling(problem::TrussProblem{xdim, T}, ginfo, einfo; u=undef) where {
     global_dofs = zeros(Int, ndofs_per_cell(dh))
     Kσ_e = zeros(T, size(Kσs[1]))
     celliteratortype = CellIterator{typeof(dh).parameters...}
-    _celliterator::celliteratortype = CellIterator(dh)
+    celliterator::celliteratortype = CellIterator(dh)
     TK = eltype(Kσs)
-    for (i,cell) in enumerate(_celliterator)
+    for (i,cell) in enumerate(celliterator)
         celldofs!(global_dofs, dh, i)
-        if TK <: Symmetric
-            Ferrite.assemble!(assembler, global_dofs, Kσs[i].data)
+        Kσ_e = TK isa Symmetric ? Kσs[i].data : Kσs[i]
+        if black[i]
+            Ferrite.assemble!(assembler, global_dofs, Kσ_e)
+        elseif white[i]
+            # if PENALTY_BEFORE_INTERPOLATION
+            px = xmin
+            # else
+            #     px = penalty(xmin)
+            # end
+            Kσ_e = px * Kσ_e
+            Ferrite.assemble!(assembler, global_dofs, Kσ_e)
         else
-            Ferrite.assemble!(assembler, global_dofs, Kσs[i])
+            px = vars[varind[i]]
+            # if PENALTY_BEFORE_INTERPOLATION
+            # px = density(penalty(vars[varind[i]]), xmin)
+            # else
+                # px = penalty(density(vars[varind[i]], xmin))
+            # end
+            Kσ_e = px * Kσ_e
```

Comment:
I am not sure if we should do this for trusses unless we are going for 0-1 designs. If the thickness is optimized then the Ksigma comes from the stress and we shouldn't multiply by the density in assembly here.